storage/streams/src/main/java/io/apicurio/registry/streams/StreamsRegistryStorage.java

```diff
@@ -411,6 +435,15 @@ public StoredArtifact getArtifact(String artifactId) throws ArtifactNotFoundExce
                        });
     }
 
+
+    @Override
+    public CompletionStage<ArtifactMetaDataDto> updateArtifactWithMetadata(String artifactId, ArtifactType artifactType, ContentHandle content, EditableArtifactMetaDataDto metaData) throws ArtifactAlreadyExistsException, RegistryStorageException {
+        return updateArtifact(artifactId, artifactType, content)
+            .thenCompose(amdd -> submitter.submitMetadata(Str.ActionType.UPDATE, artifactId, -1, metaData.getName(), metaData.getDescription(), metaData.getLabels())
```

Comment:
Perhaps extract these Submitter::submitMetadata calls into new method -- so we have this (cryptic parameters :-)) in one place.
(same for Kafka storage)

Reply:
Otoh, it won't fix the cryptic params stuff or add any extra value ... so you can leave it.